src/hivpy/simulation.py

```diff
@@ -39,44 +47,47 @@ def _update_date(self, date):
         self.output_stats["Date"][self.step] = self.latest_date
 
     def _ratio(self, subpop, pop):
-        if sum(pop) != 0:
-            return sum(subpop)/sum(pop)
+        if len(pop) != 0:
+            return len(subpop)/len(pop)
         else:
             return 0
 
-    def _update_HIV_prevalence(self, pop_data):
+    def _update_HIV_prevalence(self, pop: Population):
         # Update total HIV cases and population
-        over_15_idx = selector(pop_data, age=(operator.gt, 15))
-        HIV_pos_idx = selector(pop_data, HIV_status=(operator.eq, True))
+        over_15_idx = pop.get_sub_pop([(col.AGE, operator.ge, 15)])
```

Comment:
We're including 15 year-olds in `over_15_idx` now? Might be clearer to rename this variable to something like `15_or_over_idx`, but not a big deal as long as this is documented somewhere.